openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -0,0 +1,98 @@
+"""
+Test audit user's access to various content based on content-gating features.
+"""
+
+import ddt
+from django.http import Http404
+from django.test.utils import override_settings
+from mock import Mock, patch
+
+from courseware.access import has_access
+from courseware.access_response import IncorrectPartitionGroupError
+from course_modes.tests.factories import CourseModeFactory
+from django.test.client import RequestFactory
+from lms.djangoapps.courseware.module_render import load_single_xblock
+from openedx.features.course_duration_limits.config import CONTENT_TYPE_GATING_FLAG
+from openedx.core.djangoapps.waffle_utils.testutils import override_waffle_flag
+from student.tests.factories import AdminFactory, CourseEnrollmentFactory, UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+
+
+@override_settings(FIELD_OVERRIDE_PROVIDERS=(
+                   'openedx.features.content_type_gating.field_override.ContentTypeGatingFieldOverride',))
+@override_waffle_flag(CONTENT_TYPE_GATING_FLAG, True)
+@ddt.ddt
+class TestProblemTypeAccess(ModuleStoreTestCase):
+
+    def setUp(self):
+        super(TestProblemTypeAccess, self).setUp()
+        self.course = CourseFactory.create(run='testcourse1', display_name='Test Course Title')
+        audit_mode = CourseModeFactory.create(course_id=self.course.id, mode_slug='audit')
+        verified_mode = CourseModeFactory.create(course_id=self.course.id, mode_slug='verified')
+        self.audit_user = UserFactory.create()
+        CourseEnrollmentFactory.create(user=self.audit_user, course_id=self.course.id, mode='audit')
+        with self.store.bulk_operations(self.course.id):
+            self.overview = ItemFactory.create(
+                parent=self.course,
+                display_name='Overview'
+            )
+            self.chapter_subsection = ItemFactory.create(
+                parent=self.overview,
+                category='sequential',
+                display_name='Lesson 1'
+            )
+            self.chapter_vertical = ItemFactory.create(
+                parent=self.chapter_subsection,
+                category='vertical',
+                display_name='Lesson 1 Vertical - Unit 1'
+            )
+
+    def is_block_gated(self, block, gated):
```

Comment:
The name of this util function seems to imply that it returns a boolean. Can we rename this to something like, "assert_block_is_gated" to make the purpose more clear here?

Reply:
Can you also add a docstring to this function explaining how you're using it?